dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorArm8.java

```diff
@@ -528,4 +542,23 @@ public List<Event> visitRMWOpAndTest(RMWOpAndTest e) {
                 testOp
         );	
 	};
+
+        @Override
+	public List<Event> visitLKMMLock(LKMMLock e) {
+	Register dummy = e.getThread().newRegister(GlobalSettings.ARCH_PRECISION);
+        // Spinlock events are guaranteed to succeed, i.e. we can use assumes
+	return eventSequence(
+                newRMWLoadExclusive(dummy, e.getLock(), ARMv8.MO_ACQ),
+                newAssume(new Atom(dummy, COpBin.EQ, IValue.ZERO)),
+                newRMWStoreExclusive(e.getLock(), IValue.ONE, ARMv8.MO_RX, true)
```

Comment:
Using `Assume` avoids ctrl-dependencies. It doesn't matter, because the `load` is acquire, but maybe we should put a comment about the missing ctrl-dep.

Reply:
Following the `extractXMoFromLKMo` pattern, we should make it non-atomic

Reply:
What even is `ARMv8.MO_RX`? Does ARMv8 even have a notion of relaxed atomic? I thought there is only REL, ACQ and plain (~non-atomic/relaxed).

Reply:
Good question. The cat file does not refer to the `RX` tag. I could not find anything in arm documentation. Neither in `herd7 code` (they have something called `AArch64.N` which might simply be "non-atomic"). 

Our litmus parser creates such tag, but I think we can simply change it to `""`. I will change this in a different PR.